### PR TITLE
(docs) Add resource_types deprecation notes.

### PR DESCRIPTION
**Pull request by gguillotte**
Supports [DOC-2646](https://tickets.puppetlabs.com/browse/DOC-2646).

### CODE REVIEW

File: documentation/deprecated_features.markdown [hunk 1]
```diff
@@ -347,3 +347,27 @@ use cases.
...
+
+The `environment_classes` endpoint ignores Puppet's Ruby-based authorization methods and configuration in favor of Puppet Server's Trapperkeeper authorization. For more information, see the ["Authorization" section](./puppet-api/v3/environment_classes.html#authorization) of the environment classes API documentation.
+
+#### Context
+
+Users often rely on the `resource_types` endpoint for lists of classes and associated parameters in an environment. For such requests, the `resource_types` endpoint is inefficient and can trigger problematic events, such as [manifests being parsed during a catalog request](https://tickets.puppetlabs.com/browse/SERVER-1200).
+
+To fill these requests more efficiently and safely, Puppet Server 2.3.0 introduced the narrowly defined `environment_classes` endpoint.
```
Comment (camlow325) [inline documentation/deprecated_features.markdown]: s/fill/fulfill ?

**Comment by camlow325** [comment]
LGTM.  Just one cosmetic nitpick.

**Comment by camlow325** [comment]
👍 from me.  @gguillotte, do you think anyone else needs to review this or can I just merge it?

**Comment by gguillotte** [comment]
@camlow325 Nick is out; since this crosses over to Puppet, @jtappa could take a look. @cprice404 filed the related ticket DOC-2892.

**Comment by cprice404** [comment]
+1

**Pull request merged**